genref/README.md

```diff
@@ -0,0 +1,74 @@
+# API Reference generator
+
+This is an API reference docs generator that uses the
+[k8s.io/gengo](https://godoc.org/k8s.io/gengo) project to parse types and
+generate API documentation from it.
+
+This tool relies only on the Go source code containing the type definitions.
+For references to external type definitions, especially those from the
+`k8s.io/kubernetes` repo, you can tell the tool to generate links to the
+existing docs, which can be the GoDocs if no better option exists.
+
+## Try it out
+
+1. Clone this repository.
+
+2. Make sure you have go1.15+ installed. Run `make` to build the `genref`
+   binary:
+
+   ```
+   cd genref
+   make
+   ```
+
+3. Generate reference doc for some API types.
+
+   ```shell
+   ./genref -include kubelet-config --output output/html 
+   ```
+
+4. Visit `output/html/kubelet-config.v1beta1.html` to view the results.
+
+## Customization
+
+### Specify the package
+
+You can modify the `config.yaml` file to customize the generated API
+reference output.
+You can also use `-include` and `-exclude` flags for the `genref` binary
+to customize which package to include or exclude respectively.
+
+### Specify the output format
+
+The tool can generate HTML pages directly or Markdown files if needed.
+You can specify the output format using the `-format` flag. For example,
+
+```shell
+./genref -format markdown -include kubelet-config
+```
+
+### Customize the output
+
+The outputs are generated using GoLang templates. For HTML outputs, the
```

Comment:
nit: The tool uses Go [or GoLang] templates to generate HTML or Markdown.
The HTML templates are found under the ... subdirectory. For Markdown, the templates are ...

Reply:
Will fix this in a follow up PR, along with the improvement to include descriptions.

Reply:
Okay, this PR is not approved yet, I still get chance to fix the README.